SimG4CMS/PPS/src/PPSPixelSD.cc

```diff
@@ -0,0 +1,437 @@
+
+  for (int j = 0; j < theHC->entries() && !found; j++) {
+    PPSPixelG4Hit* aPreviousHit = (*theHC)[j];
+    if (aPreviousHit->getTrackID() == primaryID && aPreviousHit->getTimeSliceID() == tSliceID &&
+        aPreviousHit->getUnitID() == unitID) {
+      currentHit = aPreviousHit;
+      found = true;
+    }
+  }
+
+  if (found) {
+    UpdateHit();
+    return true;
+  }
+  return false;
+}
+
+void PPSPixelSD::CreateNewHit() {
+#ifdef debug
+  LogDebug("PPSSim") << "PPSPixelSD CreateNewHit for"
+                     << " PV " << currentPV->GetName() << " PVid = " << currentPV->GetCopyNo()
+                     << " MVid = " << currentPV->GetMother()->GetCopyNo() << " Unit " << unitID << "\n"
+                     << " primary " << primaryID << " time slice " << tSliceID << " For Track  "
+                     << theTrack->GetTrackID() << " which is a " << theTrack->GetDefinition()->GetParticleName();
+
+  if (theTrack->GetTrackID() == 1) {
+    LogDebug("PPSSim") << " of energy " << theTrack->GetTotalEnergy();
+  } else {
+    LogDebug("PPSSim") << " daughter of part. " << theTrack->GetParentID();
+  }
+
+  if (theTrack->GetCreatorProcess() != NULL)
+    LogDebug("PPSSim") << theTrack->GetCreatorProcess()->GetProcessName();
+  else
+    LogDebug("PPSSim") << "NO process";
+#endif
+
+  currentHit = new PPSPixelG4Hit;
+  currentHit->setTrackID(primaryID);
+  currentHit->setTimeSlice(tSlice);
+  currentHit->setUnitID(unitID);
+  currentHit->setIncidentEnergy(incidentEnergy);
+
+  currentHit->setPabs(Pabs);
+  currentHit->setTof(Tof);
+  currentHit->setEnergyLoss(Eloss);
+  currentHit->setParticleType(ParticleType);
+  currentHit->setThetaAtEntry(ThetaAtEntry);
+  currentHit->setPhiAtEntry(PhiAtEntry);
+
+  currentHit->setMeanPosition(Posizio);
+  currentHit->setEntryPoint(theEntryPoint);
+  currentHit->setExitPoint(theExitPoint);
+
+  currentHit->setParentId(ParentId);
+  currentHit->setVx(Vx);
+  currentHit->setVy(Vy);
+  currentHit->setVz(Vz);
+
+  UpdateHit();
+
+  StoreHit(currentHit);
+}
+
+void PPSPixelSD::CreateNewHitEvo() {
+  currentHit = new PPSPixelG4Hit;
+  currentHit->setTrackID(primaryID);
+  currentHit->setTimeSlice(tSlice);
+  currentHit->setUnitID(unitID);
+  currentHit->setIncidentEnergy(incidentEnergy);
+
+  currentHit->setPabs(Pabs);
+  currentHit->setTof(Tof);
+  currentHit->setEnergyLoss(Eloss);
+  currentHit->setParticleType(ParticleType);
+  currentHit->setThetaAtEntry(ThetaAtEntry);
+  currentHit->setPhiAtEntry(PhiAtEntry);
+
+  currentHit->setEntryPoint(theEntryPoint);
+  currentHit->setExitPoint(theExitPoint);
+
+  currentHit->setParentId(ParentId);
+  currentHit->setVx(Vx);
+  currentHit->setVy(Vy);
+  currentHit->setVz(Vz);
+
+  G4ThreeVector _PosizioEvo;
+  int flagAcc = 0;
+  _PosizioEvo = PosizioEvo(Posizio, Vx, Vy, Vz, Pabs, flagAcc);
+
+  if (flagAcc == 1) {
+    currentHit->setMeanPosition(_PosizioEvo);
+
+    UpdateHit();
+
+    StoreHit(currentHit);
+  }
+}
+
+G4ThreeVector PPSPixelSD::PosizioEvo(
```

Comment:
please use the English language for function and variable names, since not everyone in CMS speaks Italian (also "accettanza" below

Reply:
Removed all xxxEvo methods (it was used when there was no proton propagation and they are not needed anymore).